src/compile/data/assemble.ts

```diff
@@ -36,6 +42,79 @@ function removeUnnecessaryNodes(node: DataFlowNode) {
 
   node.children.forEach(removeUnnecessaryNodes);
 }
+
+/**
+ * Clones the subtree and ignores output nodes except for the leafs, which are renamed.
```

Comment:
It seems like you also ignore unclonable node, but I'm not sure why.  
Can you explain more?